sklearn/preprocessing/_encoders.py

```diff
@@ -258,9 +266,23 @@ def _check_infrequent_enabled(self):
         max_categories = getattr(self, "max_categories", None)
         min_frequency = getattr(self, "min_frequency", None)
         self._infrequent_enabled = (
-            max_categories is not None and max_categories >= 1
+            self._max_categories_enable_infrequent(max_categories)
         ) or min_frequency is not None
 
+    def _has_infrequent_categories(self, n_current_features, col_idx):
+        """
+        This function checks if there are any infrequent categories.
```

Comment:
I think `n_current_features` is poorly named since it really refers to the number of categories, not the number of features in `X`. Indeed, this check is really asking if the categories of this feature should be collapsed due to there being more categories than allowed by `max_categories`. A better function name might be `_has_collapsable_categories` to reflect what is really being done and rename `n_current_features` to `n_current_categories`.